crates/lang/src/codegen/dispatch/execution.rs

```diff
@@ -73,6 +73,14 @@ where
 /// Configuration for execution of ink! constructor.
 #[derive(Debug, Copy, Clone)]
 pub struct ExecuteConstructorConfig {
+    /// Yields `true` if the ink! constructor accepts payment.
+    ///
+    /// # Note
+    ///
+    /// If no ink! constructor within the same ink! smart contract
+    /// is payable then this flag will be `true` since the check
+    /// then is moved before the constructor dispatch as an optimization.
```

Comment:
Hmm, not happy about this. This possibility of `payable` being set to `true` despite no payable constructor seems to be destined for a bug. How about we reflect this information e.g. with an enum here as a wrapper around the `bool`?

Reply:
I think there was already a small bug here actually. If there were indeed no payable messages for example, we'd never hit [this check](https://github.com/paritytech/ink/blob/d70973dcf45f72bd18b705e1f24c46d756133e0e/crates/lang/src/codegen/dispatch/execution.rs#L104). Although to be fair, payment would have been denied in [call()](https://github.com/paritytech/ink/blob/d70973dcf45f72bd18b705e1f24c46d756133e0e/crates/lang/codegen/src/generator/dispatch.rs#L4444) (maybe this was the optimization alluded to in the comment?). 

Either way, this was confusing and I think it's unnecessary so I removed it.